ethToElrond/v2/ethToElrond/steps/semiIntegrated2_test.go

```diff
@@ -0,0 +1,298 @@
+			return errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return errHandler.storeAndReturnError(nil)
+	}
+	stub.WasTransferProposedOnElrondCalled = func(ctx context.Context) (bool, error) {
+		if args.failingStep == wasTransferProposedOnElrond {
+			return false, errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return args.wasTransferProposedHandler(), errHandler.storeAndReturnError(nil)
+	}
+	stub.ProposeTransferOnElrondCalled = func(ctx context.Context) error {
+		if args.failingStep == proposeTransferOnElrond {
+			return errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return errHandler.storeAndReturnError(nil)
+	}
+	stub.WasProposedTransferSignedOnElrondCalled = func(ctx context.Context) (bool, error) {
+		if args.failingStep == wasProposedTransferSigned {
+			return false, errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return args.wasProposedTransferSignedHandler(), errHandler.storeAndReturnError(nil)
+	}
+	stub.SignProposedTransferOnElrondCalled = func(ctx context.Context) error {
+		if args.failingStep == signProposedTransfer {
+			return errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return errHandler.storeAndReturnError(nil)
+	}
+	stub.IsQuorumReachedOnElrondCalled = func(ctx context.Context) (bool, error) {
+		if args.failingStep == isQuorumReached {
+			return false, errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return args.isQuorumReachedHandler(), errHandler.storeAndReturnError(nil)
+	}
+	stub.WasActionIDPerformedOnElrondCalled = func(ctx context.Context) (bool, error) {
+		if args.failingStep == wasActionIDPerformed {
+			return false, errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return args.wasActionIDPerformedHandler(), errHandler.storeAndReturnError(nil)
+	}
+	stub.PerformActionIDOnElrondCalled = func(ctx context.Context) error {
+		if args.failingStep == performActionID {
+			return errHandler.storeAndReturnError(expectedErr)
+		}
+
+		return errHandler.storeAndReturnError(nil)
+	}
+	stub.ProcessMaxRetriesOnElrondCalled = func() bool {
+		return args.maxRetriesReachedHandler()
+	}
+
+	return stub, errHandler
+}
+
+func createStateMachine(t *testing.T, executor ethToElrond.EthToElrondBridge, initialStep core.StepIdentifier) *stateMachine.StateMachineMock {
+	steps, err := CreateSteps(executor)
+	require.Nil(t, err)
+
+	sm := stateMachine.NewStateMachineMock(steps, initialStep)
+	err = sm.Initialize()
+	require.Nil(t, err)
+
+	return sm
+}
+
+func TestHappyCaseWhenLeader(t *testing.T) {
+	t.Parallel()
+
+	args := argsBridgeStub{
+		myTurnHandler:                    trueHandler,
+		isQuorumReachedHandler:           trueHandler,
+		wasActionIDPerformedHandler:      trueHandler,
+		maxRetriesReachedHandler:         falseHandler,
+		wasProposedTransferSignedHandler: falseHandler,
+		wasTransferProposedHandler:       falseHandler,
+	}
+	executor, eh := createMockBridge(args)
+	sm := createStateMachine(t, executor, ethToElrond.GettingPendingBatchFromEthereum)
```

Comment:
perhaps last parameter should be removed as all calls to createStateMachine() use ethToElrond.GettingPendingBatchFromEthereum as initial state

Reply:
I would keep this as it is because we can easily add a semi-integrated test between step 2 and step 5, for example,